helix-core/src/test/java/org/apache/helix/manager/zk/TestZkClusterManager.java

```diff
@@ -54,27 +54,28 @@
 
   @Test()
   public void testController() throws Exception {
-    System.out.println("START " + className + ".testController() at "
-        + new Date(System.currentTimeMillis()));
+    System.out.println(
+        "START " + className + ".testController() at " + new Date(System.currentTimeMillis()));
     final String clusterName = CLUSTER_PREFIX + "_" + className + "_controller";
 
     // basic test
     if (_gZkClient.exists("/" + clusterName)) {
       _gZkClient.deleteRecursively("/" + clusterName);
     }
 
-    ZKHelixManager controller =
-        new ZKHelixManager(clusterName, null, InstanceType.CONTROLLER, ZK_ADDR);
-
     try {
-      controller.connect();
-      Assert.fail("Should throw HelixException if initial cluster structure is not setup");
+      ZKHelixManager controller =
```

Comment:
This is concerning. Considering if our customer code try-catch in the same old way that we have, their try-catch logic is required to be updated as well. This is not backward compatible and may fail our customer's code. Let's talk and see if there is a way to lazily load the configs so the behavior could be kept the same.

Reply:
I changed the logic so that backward compatibility can be maintained. Basically, current logic allow constructing a ZKHelixManager without setting up the cluster. So we have to allow a default cloud config setup before cluster is set up. I think it is ok as we just set enabled to be false.